File: cxf/src/main/java/com/bigdata/rpc/webservice/original/service/BookService.java
package com.bigdata.rpc.webservice.original.service;

import com.bigdata.rpc.webservice.original.bean.Book;

import java.util.List;

public interface BookService {

    boolean delete(int id);

    boolean update(Book book);

    boolean save(Book book);

    Book getById(int id);

    List<Book> getAll();

}

File: cxf/src/main/scala/com/bigdata/rpc/webservice/cxf/service/BookService.scala
package com.bigdata.rpc.webservice.cxf.service

import javax.jws.WebService
import java.util.{List => JList}

import com.bigdata.rpc.webservice.cxf.bean.Book

@WebService
trait BookService {

  def save(book: Book): Boolean

  def delete(id: Int): Boolean

  def update(book: Book): Boolean

  def getById(id: Int): Book

  def getAll(): JList[Book]

}
